crates/storage/trie/db.rs

```diff
@@ -0,0 +1,137 @@
+use crate::error::StoreError;
+use ethereum_rust_core::rlp::{decode::RLPDecode, encode::RLPEncode};
+use ethereum_types::H256;
+use libmdbx::{
+    orm::{table, Database},
+    table_info,
+};
+
+/// Libmbdx database representing trie state
+/// It contains two tables, one mapping node references to rlp encoded nodes, and another one mapping node hashes to node references
+/// All nodes are stored in the DB and no node is ever removed
+/// Only node hahses for root nodes computed with `Trie::compute_hash` are stored
+/// Contains the next node reference to be used when storing a new node
+use super::{node::Node, node_ref::NodeRef};
+pub struct TrieDB {
+    db: Database,
+    next_node_ref: NodeRef,
+}
+
+/// RLP-encoded trie node
+pub type NodeRLP = Vec<u8>;
+/// RLP-encoded node hash
+pub type NodeHashRLP = [u8; 32];
+
+table!(
+    /// NodeRef to Node table
+    ( Nodes ) NodeRef => NodeRLP
+);
+
+table!(
+    /// NodeHash to NodeRef table
+    /// Stores root nodes which's hashes have been computed by `compute_hash`
+    ( RootNodes ) NodeHashRLP => NodeRef
+);
+
+impl TrieDB {
+    /// Opens a DB created by a previous execution or creates a new one if it doesn't exist
+    pub fn init(trie_dir: &str) -> Result<(TrieDB, Option<NodeRef>), StoreError> {
+        TrieDB::open(trie_dir).or_else(|_| TrieDB::create(trie_dir).map(|db| (db, None)))
+    }
+
+    /// Creates a new clean DB
+    pub fn create(trie_dir: &str) -> Result<TrieDB, StoreError> {
```

Comment:
instead of specifying a db location, we should be able to specify the table name. Like `WorldTrie`, `StorageTrie`. But we leave this refactor for later if you prefer (later we would only have one table instead of two)

Reply:
https://github.com/lambdaclass/ethereum_rust/issues/359